src/hotspot/share/gc/serial/tenuredGeneration.cpp

```diff
@@ -219,6 +219,18 @@ void TenuredGeneration::object_iterate(ObjectClosure* blk) {
   _the_space->object_iterate(blk);
 }
 
+void TenuredGeneration::complete_loaded_archive_space(MemRegion archive_space) {
+  TenuredSpace* space = (TenuredSpace*)_the_space;
+
+  space->initialize_threshold();
+  HeapWord* start = archive_space.start();
+  while (start < archive_space.end()) {
+    size_t word_size = _the_space->block_size(start); /// Crashes here when accessing the klass
```

Comment:
Please remove the debug comment :)
```suggestion
    size_t word_size = _the_space->block_size(start);
```